code/Model/EditableCustomRule.php

```diff
@@ -230,6 +231,63 @@ public function buildExpression()
         return $result;
     }
 
+
+    /**
+     * Determines whether the rule is satisfied, based on provided form data.
+     * Used for php validation of required conditional fields
+     *
+     * @param array $data Submitted form data
+     * @return boolean
+     * @throws LogicException Invalid ConditionOption is set for this rule.
+     */
+    public function validateAgainstFormData(array $data)
+    {
+
+        $controllingField = $this->ConditionField();
+
+        if (!isset($data[$controllingField->Name])) {
+            return false;
+        }
+
+        $valid = false;
+
+        $targetFieldValue = $this->FieldValue;
+        $actualFieldValue = $data[$controllingField->Name];
+
+        switch ($this->ConditionOption) {
+            case 'IsNotBlank':
+                $valid = ($actualFieldValue !== '');
+                break;
+            case 'IsBlank':
+                $valid = ($actualFieldValue === '');
+                break;
+            case 'HasValue':
+                $valid = ($actualFieldValue === $targetFieldValue);
+                break;
+            case 'ValueNot':
+                $valid = ($actualFieldValue !== $targetFieldValue);
+                break;
+            case 'ValueLessThan':
+                $valid = ($actualFieldValue < $targetFieldValue);
+                break;
+            case 'ValueLessThanEqual':
+                $valid = ($actualFieldValue <= $targetFieldValue);
+                break;
+            case 'ValueGreaterThan':
+                $valid = ($actualFieldValue > $targetFieldValue);
+                break;
+            case 'ValueGreaterThanEqual':
+                $valid = ($actualFieldValue >= $targetFieldValue);
+                break;
+            default:
+                throw new LogicException("Unhandled rule {$this->ConditionOption}");
+                break;
+        }
```

Comment:
Retrospective review to consider for the future:

I think this would read more cleanly if you utilised early returns:

```php
switch ($this->ConditionOption) {
    case 'IsNotBlank':
        return $actualFieldValue === '';
    case 'IsBlank':
        return $actualFieldValue !== '';
    case 'HasValue':
        return $actualFieldValue === $targetFieldValue;
    // ...
```